python-squid/src/main/java/org/sonar/python/cfg/ControlFlowGraphBuilder.java

```diff
@@ -154,6 +154,15 @@ private PythonCfgBlock build(Statement statement, PythonCfgBlock currentBlock) {
     return currentBlock;
   }
 
+  private PythonCfgBlock buildWithStatement(WithStatement statement, PythonCfgBlock successor) {
+    // To avoid FP, we assume that exception raised inside with statement will be suppressed by __exit__() method of Context Manager
+    // see https://docs.python.org/3/reference/compound_stmts.html#the-with-statement
+    exceptionTargets.push(successor);
```

Comment:
I am really wondering if this should be fixed at CFG level or rule level. As far as I understand this is not wrong at CFG level but rather on its usage ?

Reply:
I agree, I will handle this at rule level!

Reply:
Updated in 728df2df488632c606e1f0dbb85ef65b35ae97c7